### ISSUE TITLE
Main bundle access from within the SDK

### ISSUE THREAD

**Issue by antoinelamy** [zendesk/zendesk_sdk_chat_ios#225]
- SDK version: 1.4.0
- Xcode version: 10.1
- iOS or Simulator: Both
- iOS Version: All
- Swift version: 4.2

*Steps to Reproduce:*
1. Embed the `ZDCChat.framework` and `ZDCChatAPI.framework` into a dynamic framework along with their resource bundles
2. Link this dynamic framework to an app
3. In the app, start a chat session

*Actual Result:*
The app crash, not finding it's resource bundle. This crash is most likely due to the fact that the SDK look for its resource bundle in main bundle instead of its current bundle (`[NSBundle mainBundle]`).

*Expected result:*
The app does not crash and the SDK look for its resource bundle in the current bundle instead of the main one (`[NSBundle bundleForClass:]`)

**Comment by HamstringAssassin** [comment]
Hi @antoinelamy, 

Could you do me a favor and send some sample code that is demonstrating the problem?

Id suggest adding the frameworks to the app to try solve the issue. 

Thanks, 
Al

**Comment by antoinelamy** [comment]
@HamstringAssassin yeah I could work on a sample project to demonstrate the issue but sometime adding the framework to the app is not even possible if what you distribute is a framework. The fix should be fairly easy on your end, you just have to replace occurrences of `[NSBundle mainBundle]` with `[NSBundle bundleForClass:]`. More details can be found here (Dynamic Frameworks vs. Static Libraries): http://blog.cocoapods.org/Pod-Authors-Guide-to-CocoaPods-Frameworks/

**Comment by antoinelamy** [comment]
@HamstringAssassin Here is the sample app reproducing the issue. Just add an account key in `AppDelegate` and tap the start chat button. The app will crash not finding its resource bundle
[ZendeskSample.zip](https://github.com/zendesk/zendesk_sdk_chat_ios/files/2857640/ZendeskSample.zip)
.

**Comment by KillianSmith0** [comment]
Hey @antoinelamy 

Thanks for the sample app and article regarding the frameworks and libraries.. I made a quick attempt at what you suggested, but there seems to be a bit more to it. We'll have to set aside more time to look further into this. We'll keep you posted here,

Thanks,
Killian

**Comment by antoinelamy** [comment]
Thanks @KillianSmith0

**Comment by bielikb** [comment]
Hey @antoinelamy, 
you've got copy the bundles `ZDCChat` and `ZDCChatStrings` to your sample app. 
Let me know how that works, I'm attaching a screenshot.

<img width="1357" alt="screenshot 2019-02-21 at 10 27 38" src="https://user-images.githubusercontent.com/4924709/53162254-8ce19680-35c3-11e9-83b5-384ca6b54aa5.png">

**Comment by antoinelamy** [comment]
@bielikb yeah I know that work but I'm not distributing an app but a framework. It should work when the resource bundles are in the same bundle as the framework but it don't. I'm just curious, what where the issue you were having by replacing `[NSBundle mainBundle]` by `[NSBundle bundleForClass:]` with a class from the SDK?

**Comment by bielikb** [comment]
Hi @antoinelamy I filed a bug for this issue. 
We're actively developing Chat V2 at the moment, so this issue will be addressed.
Thanks a lot for providing us a Sample App!

Best,
Boris
